app/controllers/cinema_halls_controller.rb

```diff
@@ -1,4 +1,47 @@
 # frozen_string_literal: true
 
 class CinemaHallsController < ApplicationController
+	def index 
+		@cinema_halls = CinemaHall.all
+		render json: @cinema_halls
+	end
+
+	def create
+		@cinema_hall = CinemaHall.new(cinema_hall_params)
+
+		if @cinema_hall.save
+			render json: @cinema_halls,
+				status: :created, location: @cinema_hall 
+		else
+			render json: @cinema_halls.errors,
+				status: :unprocessable_entity
+		end
+	end
+
+	def show 
+		@cinema_hall = CinemaHall.find(params[:id])
+		render json: @cinema_halls
```

Comment:
the same here. `@cinema_hall` not halls as you want to return on record that you previously defined :)